test/v2/cashaccounts.js

```diff
@@ -0,0 +1,205 @@
+
+    it("return success object if account is valid", async () => {
+      req.params.account = "Jonathan"
+      req.params.number = "100"
+      const result = await lookup(req, res)
+
+      // Assert that required fields exist in the returned object.
+      assert.exists(result.identifier)
+      assert.exists(result.information)
+    })
+  })
+
+  describe("#check", () => {
+    // details route handler.
+    const { check } = cashaccountsRoute.testableComponents
+
+    it("should throw 400 if account is empty", async () => {
+      const result = await check(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(result.error, "account name can not be empty")
+    })
+
+    it("should throw 400 if number is empty", async () => {
+      req.params.account = "test"
+      const result = await check(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(result.error, "number can not be empty")
+    })
+
+    it("should throw 500 if handle is invalid", async () => {
+      req.params.account = "test"
+      req.params.number = "344a"
+      const result = await check(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(result.error, "Not a valid CashAccount")
+    })
+
+    it("should throw 500 if handle is valid but not found", async () => {
+      req.params.account = "Jonathan"
+      req.params.number = "111"
+      const result = await check(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(
+        result.error,
+        "No account matched the requested parameters."
+      )
+    })
+
+    it("return success object if account is valid", async () => {
+      req.params.account = "Jonathan"
+      req.params.number = "100"
+      const result = await check(req, res)
+
+      // Assert that required fields exist in the returned object.
+      assert.exists(result.identifier)
+      assert.exists(result.results)
+      assert.exists(result.block)
+    })
+  })
+
+  describe("#reverseLookup", () => {
+    // details route handler.
+    const { reverseLookup } = cashaccountsRoute.testableComponents
+
+    it("should throw 400 if address is empty", async () => {
+      const result = await reverseLookup(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(result.error, "address can not be empty")
+    })
+
+    it("should warn if using a bad address", async () => {
+      req.params.address = "bitcoincash:ljqgqqj64s9la9"
+      const result = await reverseLookup(req, res)
+
+      assert.hasAllKeys(result, ["error"])
+      assert.include(result.error, "Not a valid BCH address.")
+    })
+
+    it("should return results with valid address", async () => {
+      req.params.address =
+        "bitcoincash:qrhncn6hgkhljqg4fuq4px5qg74sjz9fqqj64s9la9"
+      const result = await reverseLookup(req, res)
+
+      // Assert that required fields exist in the returned object.
+      assert.exists(result.results)
+    })
+  })
+})
```

Comment:
These are great! Fantastic job on following the unit test pattern I showed you. 

The positive test cases though should have the `if()` logic to detect a `unit` `TEST` env var and mock to live call using Sinon.